src/wallet/wallet.cpp

```diff
@@ -3305,31 +3305,27 @@ bool ReserveDestination::GetReservedDestination(CTxDestination& dest, bool inter
         if (!m_spk_man->GetReservedDestination(type, internal, dest, nIndex, keypool)) {
             return false;
         }
-        vchPubKey = keypool.vchPubKey;
         fInternal = keypool.fInternal;
         address = dest;
     }
-    assert(vchPubKey.IsValid());
```

Comment:
Note to other reviewers: this assert is lost, but `GetReservedDestination` calls `ReserveKeyFromKeyPool` which throws if `!keypool.vchPubKey.IsValid()`